### PR TITLE
[TableList, PropTable] Refactors the layout

### PR DESCRIPTION
**Pull request by AkimaLunar**
# [TableList, PropTable] Refactors the layout

- Implements grid layout instead of the native table to improve the design.
- Implements small screen layout (collapsed table).

### CODE REVIEW

File: components/table-list/src/table-list/table-list.tsx [hunk 1]
```diff
@@ -1,24 +1,123 @@
...
 import { TableListProvider } from '../table-list-context';
 import { useTableListStyles } from './table-list.styles';
 import type { TableListProps } from './table-list.types';
 
+/**
+ * @todo [Extract useSize into its own package #103](https://github.com/microsoft/arbutus/issues/103)
+ * */
+type Size = { width: number; height: number };
+
+function useSize<T extends HTMLElement>(
```
Comment (mltejera) [inline components/table-list/src/table-list/table-list.tsx:25]: neat hook!

File: components/table-list/src/table-list/table-list.tsx [hunk 2]
```diff
@@ -1,24 +1,123 @@
...
 
+  const labelById = useId();
+  const describedById = useId();
+
+  // Small Screen Layout
+
+  const tableRef = useRef<HTMLTableElement | null>(null);
+  const size = useSize<HTMLTableElement>(tableRef);
+  const width = size?.width ?? 0;
+  const isCollapsed = width < 640;
```
Comment (mltejera) [inline components/table-list/src/table-list/table-list.tsx:59]: maybe consider using a const or breakpoint from fluent for this?

File: components/table-list/src/table-list/table-list.types.ts [hunk 3]
```diff
@@ -7,14 +7,24 @@ export type TableListProps = {
...
 
+  /**
+   * Accessible description for the table.
+   */
+  description?: string;
+
+  /**
+   * Table column widths.
+   */
+  columnWeights?: (number | string)[];
```
Comment (mltejera) [inline components/table-list/src/table-list/table-list.types.ts:27]: did you mean 'ColumnWidths' for this?

**Review by mltejera — APPROVED** [review]

File: components/table-list/src/table-list/table-list.types.ts (hunk 3 above)
Comment (AkimaLunar) [reply components/table-list/src/table-list/table-list.types.ts:27]: Hmm good point… It is basically the same thing as [grid-template-columns](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns), so maybe something like `columnSizing` is a better name 🤔

**Pull request merged**